contracts/AngleStrategyVoterProxy.sol

```diff
@@ -0,0 +1,180 @@
+    }
+
+    function withdraw(
+        address _gauge,
+        address _token,
+        uint256 _amount
+    ) public returns (uint256) {
+        require(strategies[_gauge] == msg.sender, "!strategy");
+        uint256 _balance = IERC20(_token).balanceOf(address(yearnAngleVoter));
+        yearnAngleVoter.safeExecute(_gauge, 0, abi.encodeWithSignature("withdraw(uint256)", _amount));
+        _balance = IERC20(_token).balanceOf(address(yearnAngleVoter)).sub(_balance);
+        yearnAngleVoter.safeExecute(_token, 0, abi.encodeWithSignature("transfer(address,uint256)", msg.sender, _balance));
+        return _balance;
+    }
+
+    function withdrawFromStableMaster(address stableMaster, uint256 amount, 
+        address poolManager, address token) external {
+        require(strategies[stableMaster] == msg.sender, "!strategy");
+
+        IERC20(token).safeTransfer(address(yearnAngleVoter), amount);
+
+        yearnAngleVoter.safeExecute(token, 0, abi.encodeWithSignature("approve(address,uint256)", stableMaster, 0));
+        yearnAngleVoter.safeExecute(token, 0, abi.encodeWithSignature("approve(address,uint256)", stableMaster, amount));
+
+        yearnAngleVoter.safeExecute(stableMaster, 0, abi.encodeWithSignature(
+            "withdraw(uint256,address,address,address)", 
+            amount,
+            address(yearnAngleVoter),
+            msg.sender,
+            poolManager
+            ));
+    }
+
+    function balanceOf(address _gauge) public view returns (uint256) {
+        return IERC20(_gauge).balanceOf(address(yearnAngleVoter));
+    }
+
+    function withdrawAll(address _gauge, address _token) external returns (uint256) {
+        require(strategies[_gauge] == msg.sender, "!strategy");
+        return withdraw(_gauge, _token, balanceOf(_gauge));
+    }
+
+    function deposit(address gauge, uint256 amount, address token) external {
+        require(strategies[gauge] == msg.sender, "!strategy");
+
+        yearnAngleVoter.safeExecute(token, 0, abi.encodeWithSignature("approve(address,uint256)", gauge, 0));
+        yearnAngleVoter.safeExecute(token, 0, abi.encodeWithSignature("approve(address,uint256)", gauge, amount));
+
+        yearnAngleVoter.safeExecute(gauge, 0, abi.encodeWithSignature(
+            "deposit(uint256)", 
+            amount
+            ));
+    }
+
+    function depositToStableMaster(address stableMaster, uint256 amount, 
+        address poolManager, address token) external {
+        require(strategies[stableMaster] == msg.sender, "!strategy");
+        
+        IERC20(token).safeTransfer(address(yearnAngleVoter), amount);
+
+        yearnAngleVoter.safeExecute(token, 0, abi.encodeWithSignature("approve(address,uint256)", stableMaster, 0));
+        yearnAngleVoter.safeExecute(token, 0, abi.encodeWithSignature("approve(address,uint256)", stableMaster, amount));
+        yearnAngleVoter.safeExecute(stableMaster, 0, abi.encodeWithSignature(
```

Comment:
Is it better to do this here rather than in the strategies, & then move sanTokens over to voter? I guess there's code sharing

Reply:
I think this links with the previous comment about setting up the stable master as a gauge. I'm currently doing all Angle interactions from the strategy proxy so it's all in the same place but no hard opinion on it - i don't see any gas savings from doing it in either place so can change it if you think it's better to keep the proxy + voter only for Angle related stuff